code/items/ammo/TTTAmmo.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+
+using Sandbox;
+
+using TTTReborn.Player;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_ammo")]
+    public abstract partial class TTTAmmo : Prop
+    {
+        /// <summary>
+        /// String definition of ammo type, should match TTTWeapon.AmmoType
+        /// </summary>
+        public virtual string Type { get; set; }
+        /// <summary>
+        /// Amount of Ammo within Entity.
+        /// </summary>
+        public virtual int Amount { get; set; }
+
+        /// <summary>
+        /// Maximum amount of ammo player can carry in their inventory.
+        /// </summary>
+        public virtual int Max { get; set; }
+
+        [Net]
+        private int CurrentAmmo { get; set; }
+        private int AmmoEntMax { get; set; }
+
+        /// <summary>
+        /// Fired when a player picks up any amount of ammo from the entity.
+        /// </summary>
+        protected Output OnPickup { get; set; }
+
+        public virtual string ModelPath => "models/ammo/ammo_buckshot.vmdl";
+
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            SetModel(ModelPath);
+            SetupPhysicsFromModel(PhysicsMotionType.Dynamic);
+
+            AmmoEntMax = Amount;
+            CurrentAmmo = Amount;
+        }
+
+        public override void Touch(Entity other)
+        {
+            base.Touch(other);
+            if (IsServer)
+            {
+                if (other is TTTPlayer player)
+                {
+                    var ammoType = Type.ToLower();
+                    var playerInv = player.Inventory as Inventory;
+
+                    if (playerInv.GetAmmoTypes().Contains(ammoType)) //Possible optimization point by caching weapons that can our ammo, rather than checking if player has our ammo.
```

Comment:
When this is about to be merged, write up an issue on github for this comment instead of leaving it as a comment.

Reply:
Will keep this comment open until ready to merge.